js/main.js

```diff
@@ -359,120 +354,114 @@ var renderComments = function (array) {
   return fragment;
 };
 
-// open popup with big photo  ------------------------------------------------
-var hidePopupPic = function () {
-  popupPic.classList.add('hidden');
+// show/hide popup with big photo  ------------------------------------------------
+var hidePopupPicture = function () {
+  popupPicture.classList.add('hidden');
 };
 
-var showPopupPic = function (item) {
+var showPopupPicture = function (item) {
   bodyModalOpen();
-  popupPic.classList.remove('hidden');
-  popupPic.querySelector('.social__comment-count').classList.add('hidden');
-  popupPic.querySelector('.comments-loader').classList.add('hidden');
-  popupPic.querySelector('.big-picture__img img').src = item.url;
-  popupPic.querySelector('.likes-count').textContent = item.likes;
-  popupPic.querySelector('.comments-count').textContent = item.comments.length;
-  popupPic.querySelector('.social__caption').textContent = item.description;
-  popupPic.querySelector('.social__comments').appendChild(renderComments(item.comments));
+  popupPicture.classList.remove('hidden');
+  popupPicture.querySelector('.social__comment-count').classList.add('hidden');
+  popupPicture.querySelector('.comments-loader').classList.add('hidden');
+  popupPicture.querySelector('.big-picture__img img').src = item.url;
+  popupPicture.querySelector('.likes-count').textContent = item.likes;
+  popupPicture.querySelector('.comments-count').textContent = item.comments.length;
+  popupPicture.querySelector('.social__caption').textContent = item.description;
+  popupPicture.querySelector('.social__comments').appendChild(renderComments(item.comments));
   document.addEventListener('keydown', onPopupEsc);
 };
 
 // hashtags validation --------------------------------------------------------
 var isHashtagHaveSharp = function (arr) {
-  var result = [];
+  var result = false;
   arr.forEach(function (item) {
     if (item.charAt(0) !== '#' && item !== '') {
-      result.push(item);
+      result = true;
     }
   });
   return result;
 };
 
 var isHashtagEmpty = function (arr) {
-  var result = [];
+  var result = false;
   arr.forEach(function (item) {
     if (item.charAt(0) === '#' && item.length === 1) {
-      result.push(item);
+      result = true;
     }
   });
   return result;
 };
 
 var isHashtagTooLong = function (arr) {
-  var result = [];
+  var result = false;
   arr.forEach(function (item) {
     if (item.length > 20) {
-      result.push(item);
+      result = true;
     }
   });
   return result;
 };
 
 var isSimilarElement = function (arr) {
-  var result = [];
+  var result = false;
   var lowercaseArr = [];
   arr.forEach(function (item) {
     lowercaseArr.push(item.toLowerCase());
   });
```

Comment:
для изменения элементов массива еще хорошо подходит `map`
```suggestion
  var lowercaseArr = arr.map(function (item) {
    return item.toLowerCase();
  });
```